handler/helpers.go

```diff
@@ -0,0 +1,127 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/klarna/eremetic/assets"
+	"github.com/klarna/eremetic/formatter"
+	"github.com/klarna/eremetic/types"
+)
+
+// getFile handles the actual fetching of file from the agent.
+func getFile(file string, task types.EremeticTask) (int, interface{}) {
```

Comment:
aha, I see. maybe it should actually be (err, io.whatever) and the "Unable to fetch upstream file" could be the message of the error object?